### DISCUSSION TITLE
Can Search support fuzzy finder on filepath?

### DISCUSSION THREAD

**Discussion by mbailey** [squidfunk/mkdocs-material#4466] [Q&A]
I'm used to opening my markdown files in bash, VSCode, Obsidian using a fuzzy finder on the file path and name.

For example, I could usually find:

    /notes/customers/examplecorp/onboarding/README.md

with:

    `nocuexonr`

I noticed MkDocs search doesn't seem to support this? Am I missing something? My fingers are used to using this to access files.

**Comment by squidfunk** [comment]
Theoretically, yes, we're using the same logic for our [icon search](https://squidfunk.github.io/mkdocs-material/reference/icons-emojis/#search). However, the behavior is fundamentally different from the document search, and I'm not sure how both should co-exist. if you want to see this in Material for MkDocs, a next step would be to write a detailed proposal/draft how you'd expect this to work and how it would integrate with existing features.

**Comment by mbailey** [comment]
Maybe a small step would be to include tokens from `location` in the search index?

I created a page called `dachshund.md` without that word in the body of the page and searching for `dachshund` turned up no results.

That's probably reasonable given the word wasn't in the body, however, boosting tokens from the file path and file name in search results would be consistent with how Google SEO works. Having the word in the filename would be a strong relevance indicator.

I had a look at search/_/index.ts but haven't yet worked out if there is an easy way to include filename and filepath in the index.

**Comment by squidfunk** [comment]
Note that we're actively considering adding this functionality as part of the new search ☺️ You may subscribe to #6307.

**Reply by mbailey** [reply]
> Note that we're actively considering adding this functionality as part of the new search ☺️ You may subscribe to #6307.

Thanks for sharing the update Martin! :-)
